lib/ansible/modules/network/dcnm/dcnm_vrf.py

```diff
@@ -37,237 +39,1042 @@
+                        if not 'ip_address' in attach or not 'vlan_id' in attach:
+                            msg =  "ip_address and vlan_id are mandatory under attach parameters"
+
+        else:
+            if state == 'merged' or state == 'overridden' or \
+                    state == 'replaced' or state == 'query':
+                msg =  "config: element is mandatory for this state"
+
+        if msg:
+            raise Exception(msg)
+
+        if self.config:
+            validated = []
+            valid_vrf, invalid_params = validate_list_of_dicts(self.config, vrf_spec)
+            for vrf in valid_vrf:
+                if vrf.get('attach'):
+                    valid_att, invalid_att = validate_list_of_dicts(vrf['attach'], att_spec)
+                    vrf['attach'] = valid_att
+                    invalid_params.extend(invalid_att)
+                validated.append(vrf)
+
+            if invalid_params:
+                msg = 'Invalid parameters in playbook: {}'.format('\n'.join(invalid_params))
+                raise Exception(msg)
+
+
+    def handle_response(self, res, op):
+
+        fail = False
+        changed = True
+
+        if res.get('ERROR'):
+            fail = True
+            changed = False
+        if op == 'attach' and 'is in use already' in str(res.values()):
+            fail = True
+            changed = False
+        if op == 'deploy' and 'No switches PENDING for deployment' in str(res.values()):
+            changed = False
+
+        return fail, changed
 
 
 def main():
     """ main entry point for module execution
     """
 
     element_spec = dict(
-        action=dict(required=True, choices=['create', 'attach', 'deploy']),
         fabric=dict(required=True, type='str'),
-        vrf_name=dict(required=True, type='str'),
-        vrf_template=dict(default='Default_VRF_Universal', type='str'),
-        vrf_extension_template=dict(default='Default_VRF_Extension_Universal', type='str'),
-        vrf_template_config=dict(default='', type='str'),
-        vrf_id=dict(type=int),
-        serial_numbers_vlans=dict(type='list'),
-        deployment=dict(default=False, type='bool')
+        config=dict(required=False, type='list'),
+        state=dict(default='merged',
+                   choices=['merged', 'replaced', 'deleted', 'overridden', 'query'])
     )
 
-    required_one_of = [['vrf_id', 'vrf_name']]
-    mutually_exclusive = [['vrf_id'],
-                          ['serial_numbers_vlans']]
+    result = dict(
+        changed=False,
+        response=''
+    )
 
     module = AnsibleModule(argument_spec=element_spec,
-                           required_one_of=required_one_of,
-                           mutually_exclusive=mutually_exclusive,
                            supports_check_mode=True)
 
-    result = dict(
-        changed=False,
-        response=dict()
-    )
+    dcnm_vrf = DcnmVrf(module)
+
+    dcnm_vrf.validate_input()
+
+    dcnm_vrf.get_want()
+    dcnm_vrf.get_have()
+
```

Comment:
Question.  If I compare the `get_want()` and `get_have()` states at this point then the dictionary structures will be identical right?

Reply:
Yeah.